.github/workflows/python-pure-wheels.yml

```diff
@@ -35,41 +35,56 @@ jobs:
         run: |
           echo "run=$SHOULD_RUN" >> $GITHUB_OUTPUT
         env:
-          SHOULD_RUN: ${{ github.event_name != 'release' || ( github.ref_type == 'tag' && startsWith(github.ref, format('refs/tags/{0}-v', matrix.target.name)) ) }}
+          SHOULD_RUN: ${{ github.event_name != 'release' || ( github.ref_type == 'tag' && startsWith(github.ref, format('refs/tags/{0}-v', matrix.target.dir)) ) }}
 
       - uses: actions/checkout@v4
         if: ${{ steps.check-tag.outputs.run == 'true' }}
       - name: Run sccache-cache
         if: ${{ steps.check-tag.outputs.run == 'true' }}
         uses: mozilla-actions/sccache-action@v0.0.5
-      - name: Install poetry
+
+
+      - name: Set up uv
         if: ${{ steps.check-tag.outputs.run == 'true' }}
-        run: pipx install poetry
-      - name: Set up Python '3.10'
+        run: curl -LsSf https://astral.sh/uv/0.3.4/install.sh | sh
+      - name: Set up Python
         if: ${{ steps.check-tag.outputs.run == 'true' }}
         uses: actions/setup-python@v5
         with:
-          python-version: '3.10'
-          cache: "poetry"
+          python-version: '3.12'
 
       - name: Build sdist and wheels
         if: ${{ steps.check-tag.outputs.run == 'true' }}
         run: |
-          cd ${{ matrix.target.name }}
-          poetry build -o ../dist
+          cd ${{ matrix.target.dir }}
+          uvx --from build pyproject-build --installer uv --outdir ../dist
```

Comment:
`uv` does not do wheel building. This is more verbose because we are just executing `pyproject-build`, while also and telling it to use `uv` internally.